src/ProposalTypesConfigurator.sol

```diff
@@ -216,21 +222,53 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
                     endIdx = endIdx + validScope.parameters[j].length;
 
                     bytes32 param = bytes32(proposedTx[startIdx:endIdx]);
-                    if (validScope.comparators[j] == Comparators.EQUAL) {
-                        bytes32 scopedParam = bytes32(validScope.parameters[j]);
-                        if (scopedParam != param) revert InvalidParamNotEqual();
+
+                    if (validScope.types[j] == SupportedTypes.UINT8) {
+                        validator.validate_uint8(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT16) {
+                        validator.validate_uint16(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT32) {
+                        validator.validate_uint32(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT64) {
+                        validator.validate_uint64(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT128) {
+                        validator.validate_uint128(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT256) {
+                        validator.validate_uint256(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
```

Comment:
we can make an entrypoint function in the `Validator` library which takes in the type, and the other paremeters and then does all of the if else checks.